sklearn/neighbors/approximate.py

```diff
@@ -309,15 +318,23 @@ def _get_radius_neighbors(self, query, max_depth, bin_queries, radius):
         total_neighbors = np.array([], dtype=int)
         total_distances = np.array([], dtype=float)
 
+        index_offsets = bin_queries // (2**self._residual)
+
         while (max_depth > self.min_hash_match and
                ratio_within_radius > threshold):
             left_mask = self._left_mask[max_depth]
             right_mask = self._right_mask[max_depth]
             candidates = []
             for i in range(self.n_estimators):
-                start, stop = _find_matching_indices(self.trees_[i],
+                start, stop = _find_matching_indices(self.trees_[i]
+                                                     [self._locations
+                                                      [i, index_offsets[i]]:
```

Comment:
+1, also to test stuff it would be great to reuse the scripts of https://github.com/erikbern/ann-benchmarks so as to display the query time vs precision for many values of `n_candidate` and `n_estimators` on the same plot along side with the annoy curve.